### PR TITLE
Support for Accessing and Verifying Notifications and Dialogs for UI Testing

### PR DESCRIPTION
**Pull request by KremnevDmitry**
See:
* #3409
* #3408

### CODE REVIEW

**Review by Flaurite — CHANGES_REQUESTED** [review]

File: jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/JmixUiTestExtension.java [hunk 1]
```diff
@@ -400,6 +406,14 @@ protected void clearViewBasePackages(ExtensionContext context) {
         }
     }
 
+    protected void clearOpenedNotifications(ExtensionContext context) {
```
Comment (Flaurite) [inline jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/JmixUiTestExtension.java:409]: Add tests for dialogs and notifications to `flowui` module.

File: jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/notification/NotificationInfo.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    protected final Notification notification;
+
+    protected String text;
+    protected String title;
+    protected String message;
+    protected Component component;
+
+    protected Notifications.Type type;
+    protected boolean closeable;
```
[the remark this answers is not in the document]
Comment (KremnevDmitry) [reply jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/notification/NotificationInfo.java:36]: Actually, not anymore :)

File: jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/FlowuiTestAssistConfiguration.java [hunk 3]
```diff
@@ -30,10 +30,13 @@
 import io.jmix.flowui.view.navigation.*;
 import org.springframework.beans.factory.config.BeanDefinition;
 import org.springframework.context.annotation.*;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
 import org.springframework.core.annotation.Order;
 
 @Configuration
 @Import({VaadinScopesConfig.class})
```
Comment (Flaurite) [inline jmix-flowui/flowui-test-assist/src/main/java/io/jmix/flowui/testassist/FlowuiTestAssistConfiguration.java:39]: Add `ComponentScan` annotation. And change all bean prefixes to `flowui` excluding existing beans.

**Review by Flaurite — APPROVED** [review]

**Comment by KremnevDmitry** [comment]
Will be merged after RC-1 release

**Pull request merged**